packages/components/src/toggle-group-control/toggle-group-control/component.tsx

```diff
@@ -20,43 +20,75 @@ import { VisualLabelWrapper } from './styles';
 
 /**
- * A utility used to animate something (e.g. an indicator for the selected option
- * of a component).
+ * A utility used to animate something in a container component based on the "offset
+ * rect" (position relative to the container and size) of a subelement. For example,
+ * this is useful to render an indicator for the selected option of a component, and
+ * to animate it when the selected option changes.
  *
- * It works by tracking the position and size (i.e., the "rect") of a given subelement,
- * typically the one that corresponds to the selected option, relative to its offset
- * parent. Then it:
+ * Takes in a container element and the up-to-date "offset rect" of the target
+ * subelement, obtained with `useTrackElementOffsetRect`. Then it does the following:
  *
- * - Keeps CSS variables with that information in the parent, so that the animation
- *   can be implemented with them.
- * - Adds a `is-animation-enabled` CSS class when the element changes, so that the
- *   target (e.g. the indicator) can be animated to its new position.
- * - Removes the `is-animation-enabled` class when the animation is done.
+ * - Adds CSS variables with rect information to the container, so that the indicator
+ *   can be rendered and animated with them. These are kept up-to-date, enabling CSS
+ *   transitions on change.
+ * - Sets an attribute (`data-subelement-animated` by default) when the tracked
+ *   element changes, so that the target (e.g. the indicator) can be animated to its
+ *   new size and position.
+ * - Removes the attribute when the animation is done.
+ *
+ * The need for the attribute is due to the fact that the rect might update in
+ * situations other than when the tracked element changes, e.g. the tracked element
+ * might be resized. In such cases, there is no need to animate the indicator, and
+ * the change in size or position of the indicator needs to be reflected immediately.
  */
-function useSubelementAnimation(
-	subelement?: HTMLElement | null,
+function useAnimatedOffsetRect(
+	/**
+	 * The container element.
+	 */
+	container: HTMLElement | undefined,
+	/**
+	 * The rect of the tracked element.
+	 */
+	rect: ElementOffsetRect,
 	{
-		parent = subelement?.offsetParent as HTMLElement | null | undefined,
 		prefix = 'subelement',
+		attribute = `${ prefix }-animated`,
 		transitionEndFilter,
 	}: {
-		parent?: HTMLElement | null | undefined;
+		/**
+		 * The prefix used for the CSS variables, e.g. if `prefix` is `selected`, the
+		 * CSS variables will be `--selected-top`, `--selected-left`, etc.
+		 * @default 'subelement'
+		 */
 		prefix?: string;
+		/**
+		 * The name of the data attribute used to indicate that the animation is in
+		 * progress.
+		 * @default `${ prefix }-animated`
+		 */
+		attribute?: string;
+		/**
+		 * A function that is called with the transition event and returns a boolean
+		 * indicating whether the animation should be stopped. The default is to
+		 * always stop the animation.
```

Comment:
Let's make it clear that the function should return `true` when the `transitionend` event should be used to stop the animation, and instead return `false` when the `transitionend` event should be ignored.

Maybe adding a default value of `() => true` can also help clarify how it works.

Reply:
Got it. What about having a default value?

Reply:
Alright 👍